controllers/pkg/reconcilers/repository/README.md

```diff
@@ -0,0 +1,39 @@
+# repository controller
+
+The repo controller is a k8s controller acting on repository.infra.nephio.org and handles the lifecycle of the repository in gitea.
+
+For each repo CR the repo-controller handles the lifecycle of the repository in gitea. Updates are limited based on gitea's capabilities, so it is better to delete and recreate the repo or handle updates directly in gitea.
+
+## implementation
+
+The implementation assumes the repo-controller runs in the same cluster as the gitea server. Based on the environment variables we help the controller to connect to the gitea server.
```

Comment:
Post R1, we should consider Gitea running anywhere. Overall we want the model to work with Gitea, GitHub, GitLab, and other Git providers.

Reply:
agreed. there is 2 things needed for this in my view:
1. a dns setup that allows to get to a URL from anywhere. So with this we can reach the server uniformly.
2. we need to come up with a means to represent the gift provider. There is multiple options here.
- introduce a type in the CR ( I have the feel the git semantics are uniform)
- introduce a provider with a dedicated implementation
- other?